integration/templates/react-vite/src/custom-user-button/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { UserButton } from '@clerk/clerk-react';
+import { useContext } from 'react';
+import { PageContext, PageContextProvider } from '../PageContext.tsx';
+
+function Page1() {
+  const { counter, setCounter } = useContext(PageContext);
+
+  return (
+    <>
+      <h1 data-page={1}>Page 1</h1>
+      <p data-page={1}>Counter: {counter}</p>
+      <button
+        data-page={1}
+        onClick={() => setCounter(a => a + 1)}
+      >
+        Update
+      </button>
+    </>
+  );
+}
+
+export default function Page() {
+  return (
+    <PageContextProvider>
+      <UserButton>
+        <UserButton.UserProfilePage
+          label={'Page 1'}
+          labelIcon={<p data-label-icon={'page-1'}>🙃</p>}
+          url='page-1'
+        >
+          <Page1 />
+        </UserButton.UserProfilePage>
+        <UserButton.UserProfilePage label={'security'} />
+        <UserButton.UserProfilePage
+          label={'Page 2'}
+          labelIcon={<p data-label-icon={'page-2'}>🙃</p>}
+          url='page-2'
+        >
+          <h1>Page 2</h1>
+        </UserButton.UserProfilePage>
+        🌐
```

Comment:
Is this purposefully in between items here?

Reply:
Yes. Although i can't think of a way we can break this. But be safe than sorry.

Reply:
Got it, if you are actively testing that the component doesn't render this, it'd be helpful to know that in some fashion, wrap in a button that has a text documenting that it shouldn't be rendered vs just having an emoji dropped here. Non-blocking thought.